### DISCUSSION TITLE
Can hilo-mpc train the ODE model parameters according to the actual value of the ODE model parameters without knowing

### DISCUSSION THREAD

**Discussion by Snakebinp** [hilo-mpc/hilo-mpc#19] [Q&A]
I studied example: nmpc_hybrid_bio.ipynb. Then I found that the ANN model fits 'mu', 'Rs', 'Rfp' according to 'S', 'I' of the original data. But in general, the specific reaction rate of the fermentation process is unknown. How should I use hilo-mpc to train the kinetic parameters based on the existing data (including t, X, S, P, I, V, Feed, etc.) model, and then combine the two to build a Hybrid model.

**Comment by brunomorampc** [comment]
Do you mean that usually you don't know which variable enters the reaction rate right? In this case you can add all the possible variables as  features and then use some feature reduction techniques if you want to reduce the number of features. 
Otherwise you can just train the NN with all features exactly on the same way you trained on S and I.

Does this answer your question?

**Reply by Snakebinp** [reply]
Thank you very much for your reply. I don't mean to say that I'm not sure which variables to choose as the feature. What I mean is that the true value of label for neural network training is unknown.  For example, I get the batch fermentation biomass ode: dx/dt = mu*x, and I have the true data of X and t but no mu. In this case, can I also train an ANN model that outputs mu through hilo-mpc to build a hybrid model with the mechanism model?

**Reply by brunomorampc** [reply]
I see. 
So you have two options. The trivial one is to  compute the rates from the concentration measurements to get the estimated rates and then use those as labels. The other one is to use physics informed learning which implicitly learn the rates without never computing it. I did implement this method in the past using casadi. but i don't remember if we have this implemented in Hilo-mpc. @jpohlodek do we have it implemented?

**Reply by brunomorampc** [reply]
I asked @jpohlodek , it turns out that physics informed learning is under development and will be available sometime in the future but we don't know yet when:(

**Reply by Snakebinp** [reply]
OK. Thanks anyway
